src/components/Common/auth/auth.tsx

```diff
@@ -0,0 +1,93 @@
+import type { PropsWithChildren } from "react";
+import * as Styled from "./auth.styles";
+
+const Header = ({ text }: { text: string }) => {
+  return <Styled.Header>{text}</Styled.Header>;
+};
+
+// ... 부분
+const BodyTopsDots = () => {
+  return (
+    <Styled.BodyTopDots>
+      <Styled.Dot>•</Styled.Dot>
+      <Styled.Dot>•</Styled.Dot>
+      <Styled.Dot>•</Styled.Dot>
+    </Styled.BodyTopDots>
+  );
+};
+
+// form 윗부분
+const BodyTop = ({ step }: { step: number }) => (
+  <Styled.BodyTop>
+    <Styled.BodyTopWrapper>
+      <Styled.BodyTopNumber opacity={step === 1 ? 1 : 0.6}>
+        1
+      </Styled.BodyTopNumber>
+      <BodyTopsDots />
+      <Styled.BodyTopNumber opacity={step === 2 ? 1 : 0.6}>
+        2
+      </Styled.BodyTopNumber>
+    </Styled.BodyTopWrapper>
+    <Styled.BodyTopText>
+      {step === 1 ? "기본 정보를 입력해주세요" : "학교 이메일을 인증해주세요"}
+    </Styled.BodyTopText>
+  </Styled.BodyTop>
+);
+
+// input
+const CustomInput = ({
+  type,
+  name,
+  required,
+  width,
+  ...props
+}: {
+  type: string;
+  name: string;
+  required: boolean;
+  width?: string;
+  props?: React.InputHTMLAttributes<HTMLInputElement>;
+}) => {
```

Comment:
타입 선언은 types.ts에 몰아서 정리하는 게 좋습니다.
타입은 참고하는 용도이기 때문에 tsx 파일에 기재해두면 가독성을 해치는 영향을 주더라구요
또한 기재한 타입들은 모두 기본 attribute이기 때문에 이를 활용하면 더 좋을 것 같습니다.
```suggestion

const Foo = (inputProps: InputHTMLAttributes<HTMLInputElement>) => {
  return <input {...inputProps} />
}
```

Reply:
감사합니다! 훨씬 깔끔하고 가독성이 좋습니다